src/lib/containers/table/SynapseTable.tsx

```diff
@@ -391,13 +405,39 @@ export default class SynapseTable extends React.Component<
     )
   }
 
-  private renderDropdownColumnMenu = (headers: SelectColumn[]) => {
+  /* 
+    The ColumnSelection dropdown state is held in SynapseTable because the EllipsisDropdown has
+    an option to open the dropdown, 'show columns'
+  */
+  public onToggleColumnSelectionShow(
+    _show: boolean,
+    _event: React.SyntheticEvent<Dropdown<'div'>, Event>,
+    metadata: any,
+  ) {
+    // Any click event for the Dropdown will close the dropdown (assuming its open), so we have
+    // to handle the onToggle event and manually manage the dropdown open state. If metadata
+    // is defined the event occuring is inside the dropdown which we then want to keep open, otherwise
+    // we close it.
+    if (metadata.source) {
```

Comment:
that change works! though that seems like odd behavior to me because the component's display should be controlled by the showColumnSelection value.. so if that line gets removed then showColumnSelection is never set to true, but it still opens and closes as expected. I'd like to keep it as is because its more explicit.